`JDBC_EXTENSIONS` compilation option doesn't work anymore

Hi @michael-o ,

in #465 you added support for `JDBC_EXTENSIONS` as a compilation option, however it doesn't seem to work anymore, so [that never works](https://github.com/xerial/sqlite-jdbc/blob/3d04d7df0c89240add2c92189adb30b6cb7e6ae0/src/test/java/org/sqlite/ExtensionTest.java#L72).

I tracked it down to [this not doing anything](https://github.com/xerial/sqlite-jdbc/blob/a6d63ee9cd740750e338a3604b7d542989d5ceef/Makefile#L69), maybe because the source file changed and the perl line doesn't work as expected anymore?

If you have some time, do you want to have a look at this?

Thanks

I have done now the following (all on FreeBSD):
```
$ sqlite3 --version
3.37.2 2022-01-06 13:25:41 872ba256cbf61d9290b571c0e6d82a20c224ca3ad82971edc46b29818d5dalt1
$ git rm -r src/main/resources/org/sqlite/native
$ gmake native SQLITE_OBJ=/usr/local/lib/libsqlite3.so SQLITE_HEADER=/usr/local/include/sqlite3.h JAVAC=javac JAVA=java CC=cc
## This version now defitively does not contain `JDBC_EXTENSIONS`
$ mvn clean verify
```

output:
```
[INFO] Running org.sqlite.ExtensionTest
[WARNING] Tests run: 3, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.015 s - in org.sqlite.ExtensionTest
```
as expected
```
[INFO] Running org.sqlite.DBMetaDataTest
[ERROR] Tests run: 38, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.151 s <<< FAILURE! - in org.sqlite.DBMetaDataTest
[ERROR] version  Time elapsed: 0.005 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: db version ==> expected: <3.39.2> but was: <3.37.2>
        at org.sqlite.DBMetaDataTest.version(DBMetaDataTest.java:1360)

[INFO] Running org.sqlite.util.OSInfoTest
Aug 10, 2022 1:38:35 PM org.sqlite.util.OSInfoTest displayOSInfo
INFORMATION: Hardware name: amd64

Aug 10, 2022 1:38:35 PM org.sqlite.util.OSInfoTest displayOSInfo
INFORMATION: OS name: FreeBSD
Aug 10, 2022 1:38:35 PM org.sqlite.util.OSInfoTest displayOSInfo
INFORMATION: Architecture name: x86_64
```
as expected
```
[INFO] Running org.sqlite.MathFunctionsTest
[ERROR] Tests run: 27, Failures: 2, Errors: 4, Skipped: 0, Time elapsed: 0.015 s <<< FAILURE! - in org.sqlite.MathFunctionsTest
[ERROR] square  Time elapsed: 0 s  <<< ERROR!
org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (no such function: square)
        at org.sqlite.MathFunctionsTest.square(MathFunctionsTest.java:225)

[ERROR] cot  Time elapsed: 0.001 s  <<< ERROR!
org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (no such function: cot)
        at org.sqlite.MathFunctionsTest.cot(MathFunctionsTest.java:120)

[ERROR] log  Time elapsed: 0 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <0.693147180559945> but was: <0.30102999566398114>
        at org.sqlite.MathFunctionsTest.log(MathFunctionsTest.java:163)

[ERROR] atn2  Time elapsed: 0.001 s  <<< ERROR!
org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (no such function: atn2)
        at org.sqlite.MathFunctionsTest.atn2(MathFunctionsTest.java:80)

[ERROR] coth  Time elapsed: 0.001 s  <<< ERROR!
org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (no such function: coth)
        at org.sqlite.MathFunctionsTest.coth(MathFunctionsTest.java:128)

[ERROR] log10  Time elapsed: 0.001 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <1.0> but was: <0.9999999999999999>
        at org.sqlite.MathFunctionsTest.log10(MathFunctionsTest.java:171)
```
well, someone did not check for `JDBC_EXTENSIONS`.

So this needs improvement.

Is that sufficient for you?

--- a/Makefile
+++ b/Makefile
@@ -66,7 +66,7 @@ $(SQLITE_OUT)/sqlite3.o : $(SQLITE_UNPACKED)
 	    $(SQLITE_SOURCE)/sqlite3.c > $(SQLITE_OUT)/sqlite3.c.tmp
 # register compile option 'JDBC_EXTENSIONS'
 # limits defined here: https://www.sqlite.org/limits.html
-	perl -p -e "s/#if SQLITE_LIKE_DOESNT_MATCH_BLOBS/  \"JDBC_EXTENSIONS\",\n#if SQLITE_LIKE_DOESNT_MATCH_BLOBS/;" \
+	perl -p -e "s/^(static const char \* const sqlite3azCompileOpt.+)$$/\1\n\n\/* This has been automatically added by sqlite-jdbc *\/\n  \"JDBC_EXTENSIONS\",/;" \
 	    $(SQLITE_OUT)/sqlite3.c.tmp > $(SQLITE_OUT)/sqlite3.c
 	cat src/main/ext/*.c >> $(SQLITE_OUT)/sqlite3.c
 	$(CC) -o $@ -c $(CCFLAGS) \

